TensorLib/TensorElement.lean

```diff
@@ -0,0 +1,26 @@
+/-
+Copyright (c) 2024 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: John Tristan, Paul Govereau, Sean McLaughlin
+-/
+
+namespace TensorLib
+
+-- Since one goal of TensorLib is to support ML frameworks and languages
+-- like Triton and NKI, we will need to support a menagerie of data types.
+-- Triton and NKI supports signed and unsigned ints (8/16/32/64) and
+-- several floating point types and sizes (float, bfloat, tfloat).
+class TensorElement (a: Type) [Add a][Sub a][Mul a][Neg a] /- etc. -/ where
```

Comment:
use section, variables?

Reply:
Tried this. For some reason Lean doesn't abstract away the type variable in a section. E.g. in 
```
section TensorElement

variable (a: Type)
variable [Add a][Sub a][Mul a][Neg a]

class TensorElement where

end TensorElement

instance : TensorElement Int where
```
the instance doesn't typecheck. Indeed, when I `#check TensorElement` I get `Type` rather than `Type -> instances... -> Type` which is what I expected. Going to leave it as is for now until I figure out how sections work with classes.